java/ql/lib/semmle/code/Location.qll

```diff
@@ -69,21 +69,8 @@ class Top extends @top {
   predicate hasLocationInfo(
     string filepath, int startline, int startcolumn, int endline, int endcolumn
   ) {
-    this.hasLocationInfoAux(filepath, startline, startcolumn, endline, endcolumn)
-    or
-    exists(string outFilepath, int outStartline, int outEndline |
-      this.hasLocationInfoAux(outFilepath, outStartline, _, outEndline, _) and
-      hasSmapLocationInfo(filepath, startline, startcolumn, endline, endcolumn, outFilepath,
-        outStartline, outEndline)
-    )
-  }
-
-  private predicate hasLocationInfoAux(
-    string filepath, int startline, int startcolumn, int endline, int endcolumn
-  ) {
-    exists(File f, Location l | fixedHasLocation(this, l, f) |
-      locations_default(l, f, startline, startcolumn, endline, endcolumn) and
-      filepath = f.getAbsolutePath()
+    exists(Location l | fixedHasLocation(this, l, _) |
```

Comment:
I think this change introduces some inconsistency. There are going to be `Top` elements where `getFile` and `hasLocationInfo` reports different files. `getFile` returning the delomboked file, and `hasLocationInfo` returning the original file containing lombok annotations.
```
private predicate test(Top t, string f1, string f2) {
  t.getFile().getAbsolutePath() = f1 and
  t.hasLocationInfo(f2, _, _, _, _) and
  f1 != f2
}
```